### ISSUE TITLE
[TF2] [Bug] [Art] Procedural Bones shrink model when rotation exceeds 30 degress

### ISSUE THREAD

**Issue by AtomicTEM** [ValveSoftware/Source-1-Games#5199]
https://github.com/ValveSoftware/Source-1-Games/assets/51938157/7ace04fc-7a74-43d4-b2c1-a5ece6a5a8bd

When ever a procedural bone is setup to rotate 90 degrees whenever its driver bone rotates 90, once the procedural bones passes the rotational threshold of 30 degree, the Vertices/model associated to the procedural bone will begin to shrink. The greater the rotation passed 30 degrees, the more the model shrinks.

the .vrd file for the model used teh linked video is teh following:
<helper> hlp_barrel weapon_bone_1 weapon_bone_1 vm_weapon_bone_3
<basepos> 0 8.330828 3.264407
<trigger> 90  0 0 0  0 0 0  0 0 0
<trigger> 90   0 0 -90    0 0 -90   0 0 0

Attempts to use more triggers to reduce shrinking has been attempted, and had some success, but this never fully removed the shrinking and added lag to the rotation, which is some cases would also be badly interpolated causing unintended movement.

**Comment by gabrielwoj** [comment]
Does that happen in regular gameplay too? Because HLMV does have some issues related to bones, more specifically the hlp_forearm ones on Scout, Demoman and Heavy.

Also, you are using HLMV++, which is a modification of HLMV, and not the actual HLMV. Regardless, an in-game preview would probably be more ideal. Perhaps go to your own LAN server, put `sv_cheats 1` and `host_timescale 0.1`, and record your game. `host_timescale` makes the game slow-mo.

**Comment by AtomicTEM** [comment]
> Does that happen in regular gameplay too? Because HLMV does have some issues related to bones, more specifically the hlp_forearm ones on Scout, Demoman and Heavy.
> 
> Also, you are using HLMV++, which is a modification of HLMV, and not the actual HLMV. Regardless, an in-game preview would probably be more ideal. Perhaps go to your own LAN server, put `sv_cheats 1` and `host_timescale 0.1`, and record your game. `host_timescale` makes the game slow-mo.

Yes it also happens in regular gameplay, and I have the slow-mo host_timescale to verify, I simply use HLMV++ to record.

**Comment by garrysmod2** [comment]
Hey everyone, I ran into a similar issue, possibly the exact same issue (I'm not sure) which is why I want to notify y'all

https://github.com/Facepunch/garrysmod-issues/issues/6999#issuecomment-5363210081

I made a proto fix, for SDK 2013 at least. Please if you are able to, test your custom Iron bomber model & animations with this code patch applied, and let me know if this fixes it
I say proto because...it fixes my ragdoll, but I don't know the full extent of that fix & any implications, it was done on the spot to give Rubat an idea
